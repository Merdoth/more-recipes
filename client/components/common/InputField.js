import React from 'react';


const InputField = ({
  type, name, placeholder, error, value, onChange, label
}) => (
    <div
        className={ `form-group ${error ? 'has-error' : ''}` }
    >
        <label
            htmlFor= { name }
            className="control-label" >

            { label }

        </label>

        <input
            id=""
            value={ value }
            onChange={ onChange }
            type= { type }
            name= { name }
            className="form-control"
            placeholder= { placeholder }
            required=""
            autoFocus=""
        />

        <br />
    </div>
);

export default InputField;
