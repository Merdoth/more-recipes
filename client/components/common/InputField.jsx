import React from 'react';
import { PropTypes } from 'prop-types';

const InputField = ({
  type,
  name,
  placeholder,
  value,
  onChange,
  label,
  required
}) => (
  <div className="input-field buttonb">
    <label htmlFor={name} className="control-label">
      {label}
    </label>

    <input
      type={type}
      name={name}
      placeholder={placeholder}
      value={value}
      onChange={onChange}
      className="form-control"
      required={required}
      style={{
        border: '0',
        outline: '0',
        background: 'transparent',
        borderBottom: '1px solid black',
        borderRadius: '3px'
      }}
    />
  </div>
);

InputField.propTypes = {
  name: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  placeholder: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  required: PropTypes.bool.isRequired
};

InputField.defaultProps = {
  type: 'text'
};

export default InputField;
