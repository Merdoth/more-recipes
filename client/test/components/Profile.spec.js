import React from 'react';
import { shallow } from 'enzyme';
import expect from 'expect';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import ConnectedProfile, { Profile } from '../../components/Profile.jsx';
import InputField from '../../components/common/InputField.jsx';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

/* global jest */
let props;
const setup = () => {
  props = {
    getOneUser: () => { },
    updateUserProfile: () => jest.fn,
  };
  return shallow(<Profile {...props} />);
};

describe('Profile Component snapshot', () => {
  it('it should render the right amount of elements', () => {
    const wrapper = setup();
    expect(wrapper.find(InputField)).toHaveLength(3);
    expect(wrapper).toMatchSnapshot();
  });

  it('should return profile', () => {
    const wrapper = setup();
    wrapper.setState({
      fullName: 'Samuel Uyali',
      userName: 'Samuel',
      email: 'samuel.uyali@gmail.com',
    });
    expect(wrapper).toMatchSnapshot();
  });
});

describe('onSubmit()', () => {
  it('should update profile to the state', () => {
    const event = {
      preventDefault: jest.fn(),
      updateUserProfile: jest.fn()
    };
    const wrapper = setup();
    const button = wrapper.find('.profileButton');
    button.simulate('click', event);
  });
});

describe('componentWillReceiveProps()', () => {
  it('should call componentWillReceiveProps()', () => {
    const wrapper = setup();
    wrapper.setProps({ user: { fullName: 'test' } });
    expect(wrapper.state('fullName')).toEqual('test');
  });
});

describe('onChange()', () => {
  it('should set fullName to state when input values changes', () => {
    const event = {
      target: { name: 'fullName', value: '' }
    };
    const wrapper = setup();
    const fullNameInput = wrapper.find('#fullName');

    event.target.value = 'Sarah Gigs';
    fullNameInput.simulate('change', event);

    expect(wrapper.instance().state.fullName).toBe('Sarah Gigs');
  });

  it('should set userName state when input values changes', () => {
    const event = {
      target: { name: 'userName', value: '' }
    };
    const wrapper = setup();
    const userNameInput = wrapper.find('#userName');

    event.target.value = 'Sarah';
    userNameInput.simulate('change', event);

    expect(wrapper.instance().state.userName).toBe('Sarah');
  });

  it('should set email state when input values changes', () => {
    const event = {
      target: { name: 'email', value: '' }
    };
    const wrapper = setup();
    const emailInput = wrapper.find('#email');

    event.target.value = 'sarah.gigs@gmail.com';
    emailInput.simulate('change', event);

    expect(wrapper.instance().state.email).toBe('sarah.gigs@gmail.com');
  });
});

describe('Connected Profile component', () => {
  it('tests that the component successfully rendered', () => {
    const store = mockStore({
      getOneUser: {
        users: {
          user: { id: 1 }
        }
      },
    });
    const wrapper = shallow(<ConnectedProfile store={store} />);
    expect(wrapper.length).toBe(1);
  });
});
