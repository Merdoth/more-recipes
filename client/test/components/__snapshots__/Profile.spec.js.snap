// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Component snapshot it should render the right amount of elements 1`] = `
<div>
  <form
    className="profile"
  >
    <h1
      id="profile-header"
    >
      Profile
    </h1>
    <InputField
      id="fullName"
      label="Fullname:   "
      name="fullName"
      onChange={[Function]}
      placeholder="null"
      required={true}
      type="text"
      value=""
    />
    <InputField
      id="userName"
      label="Username:   "
      name="userName"
      onChange={[Function]}
      placeholder="null"
      required={true}
      type="text"
      value=""
    />
    <InputField
      id="email"
      label="Email:    "
      name="email"
      onChange={[Function]}
      placeholder="null"
      required={true}
      type="text"
      value=""
    />
    <Button
      className="btn btn-lg profileButton"
      name="Update Profile"
      onClick={[Function]}
      type="submit"
    />
  </form>
</div>
`;

exports[`Profile Component snapshot should return profile 1`] = `
<div>
  <form
    className="profile"
  >
    <h1
      id="profile-header"
    >
      Profile
    </h1>
    <InputField
      id="fullName"
      label="Fullname:   "
      name="fullName"
      onChange={[Function]}
      placeholder="null"
      required={true}
      type="text"
      value="Samuel Uyali"
    />
    <InputField
      id="userName"
      label="Username:   "
      name="userName"
      onChange={[Function]}
      placeholder="null"
      required={true}
      type="text"
      value="Samuel"
    />
    <InputField
      id="email"
      label="Email:    "
      name="email"
      onChange={[Function]}
      placeholder="null"
      required={true}
      type="text"
      value="samuel.uyali@gmail.com"
    />
    <Button
      className="btn btn-lg profileButton"
      name="Update Profile"
      onClick={[Function]}
      type="submit"
    />
  </form>
</div>
`;
